sysconfig/bb5/users/compilers.yaml

```diff
@@ -0,0 +1,92 @@
+compilers:
+- compiler:
+    environment: {}
+    extra_rpaths: []
+    flags: {}
+    modules: []
+    operating_system: rhel7
+    paths:
+      cc: /usr/lib64/ccache/clang
+      cxx: /usr/lib64/ccache/clang++
+      f77: null
+      fc: null
+    spec: clang@3.4.2
+    target: x86_64
+- compiler:
+    environment: {}
+    extra_rpaths: []
+    flags: {}
+    modules: []
+    operating_system: rhel7
+    paths:
+      cc: /usr/lib64/ccache/gcc
+      cxx: /usr/lib64/ccache/g++
+      f77: /usr/bin/gfortran
+      fc: /usr/bin/gfortran
+    spec: gcc@4.8.5
+    target: x86_64
+- compiler:
+    environment: {}
+    extra_rpaths: []
+    flags: {}
+    modules: []
+    operating_system: rhel7
+    paths:
+      cc: /gpfs/bbp.cscs.ch/apps/compilers/install/intel-parallel-studio-cluster.2018.1/compilers_and_libraries_2018.1.163/linux/bin/intel64/icc
+      cxx: /gpfs/bbp.cscs.ch/apps/compilers/install/intel-parallel-studio-cluster.2018.1/compilers_and_libraries_2018.1.163/linux/bin/intel64/icpc
+      f77: /gpfs/bbp.cscs.ch/apps/compilers/install/intel-parallel-studio-cluster.2018.1/compilers_and_libraries_2018.1.163/linux/bin/intel64/ifort
+      fc: /gpfs/bbp.cscs.ch/apps/compilers/install/intel-parallel-studio-cluster.2018.1/compilers_and_libraries_2018.1.163/linux/bin/intel64/ifort
+    spec: intel@18.0.1
+    target: x86_64
+- compiler:
+    environment: {}
+    extra_rpaths: []
+    flags: {}
+    modules: []
```

Comment:
You may have already told me the answer to this question - but why don't we here use the spack-generated module?

Reply:
We can add compiler modules in config as well. But spack installed compilers don't need LD_LIBRARY_PATH or anything.

Another caveat adding module is polluting environment : for example, for intel compiler, if we add intel-parallel-studio, it brings all the world! (INCLUDE path for intel-mpi, mlk and everything). This causes problem while using intel compiler with HPE-MPI (because you also get intel-mpi include path and so on).

So I just used compiler paths.

Reply:
OK, make sense.